### DISCUSSION TITLE
Welcome to DL201 Discussions!

### DISCUSSION THREAD

**Discussion by jamescavanagh** [unpackAI/DL201#3] [Announcements]
Here is a good place to ask questions.

**Comment by Jing-Pan-China** [comment]
Hi, I have a question about missing value filling. How can I fill the missing value of variable "Weight" using its average value based on "Team". For example, if "Team" finland miss some values of "Weight", I can use the average weight value of Finland to fill the missing value who is Finnish. The whole dataset is here: https://www.kaggle.com/heesoo37/120-years-of-olympic-history-athletes-and-results
![WechatIMG350](https://user-images.githubusercontent.com/91070643/156119248-10f2018c-5fe1-40db-a0ed-c31d43b24537.jpeg)

**Reply by jamescavanagh** [reply]
Well, it really depends on the data. If there are other features which correlate with weight, such as height, you can make a model that will make a prediction for the weight based off of the height, ect. 

In this case, it is a good first start. 

Note, if most of the columns are also missing for that sample, then it would be a good idea to delete it

**Reply by Jing-Pan-China** [reply]
Hi, James. Thanks for your answer! I like your way to predict the weight based on the height.  

But if I want to use the mean value about weight of a  specific country to fill the  missing value, I still do not know how to use pandas to finish it. I have already known that I can use mean value about weight of all countries to fill the missing value  with the following  code
![哈哈哈](https://user-images.githubusercontent.com/91070643/156154596-7121a56a-8bb2-4598-940a-4d2b2a43db49.jpeg)
.

**Reply by jamescavanagh** [reply]
Ok, in this case, it can be done using pandas.groupby() to take the mean height of each country, then mapping the mean onto the missing values

**Reply by jamescavanagh** [reply]
Step one would be using .group_by
https://chrisalbon.com/code/python/data_wrangling/pandas_apply_operations_to_groups/

to get a series with the names

step 2 would be use a map
https://chrisalbon.com/code/python/data_wrangling/pandas_map_values_to_values/
